Data in Tables and Intro to SQL (#1)
Ada-Developers-Academy/core-unit-2: intro-to-sql/retrieving-records.md

----------+-----------------
        1 | Red Java
        2 | Postmodern Been
        3 | American Select
        4 | Wake-up Utopia
(4 rows)
```

#### Selecting All Columns

This SQL statement

```sql
SELECT * FROM media;
```

produces this output:

```
 media_id | category |      title      |    creator     | publication_year |           description_text
----------+----------+-----------------+----------------+------------------+---------------------------------------
        1 | book     | Red Java        | Jewel Koss     | 1971             | Nisi ducimus reprehenderit molestiae.
        2 | book     | Postmodern Been | Glen Senger    | 1929             | Dolore repellendus doloribus maiores.
        3 | movie    | American Select | Xuan Pouros    | 1997             | Corporis id voluptatem fuga.
        4 | movie    | Wake-up Utopia  | Buena Shanahan | 1930             | Fuga corrupti ducimus animi.
```

### Examples

Read through these example SQL statements. For each example, read the code and answer:

1. What is the name of the table?
1. Which columns are we selecting?
1. Which pieces of syntax are unfamiliar?

```sql
SELECT author_name FROM authors;
```

> This statement selects the `author_name` column in the `authors` table.

Reviewer: ```suggestion
<details style="max-width: 700px; margin: auto;">
  <summary>Answer</summary>

1. The name of the table is `authors`
1. The selected column is `author_name`

</details>
```

```sql
SELECT * FROM drivers;
```

> This selects all columns in the `drivers` table.

Reviewer: ```suggestion
<details style="max-width: 700px; margin: auto;">
  <summary>Answer</summary>

1. The name of the table is `drivers`
1. All columns in the table are selected

</details>
```

```sql
SELECT id, category, title FROM media;
```

> This selects the `id`, `category`, and `title` columns in the `media` table.

Reviewer: ```suggestion
<details style="max-width: 700px; margin: auto;">
  <summary>Answer</summary>

1. The name of the table is `media`
2. We select 3 columns: `id`, `category`, and `title`

</details>
```
